File: Bai_100/main.cpp
#include<bits/stdc++.h>
using namespace std;

bool isPrime(int n){
    if(n <= 1) return false;
    if(n <= 3) return true;
    if(n % 2 == 0 || n % 3 == 0) return false;
    for(int i = 5; i * i <= n; i += 6)
        if (n % i == 0 || n % (i + 2) == 0) return false;
    return true;
}

bool ThreeDivisors(int n){
    int root = sqrt(n);
    return root * root == n && isPrime(root);
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int n, cnt = 0; cin >> n;
    for(int i = 2; i * i <= n; ++i)
        if(ThreeDivisors(i * i))
            cnt++;
    cout << cnt;
    return 0;
}

File: Bai_100/main.py
def isPrime(n):
    if n <= 1: return False
    if n <= 3: return True
    if n % 2 == 0 or n % 3 == 0: return False
    for i in range(5, int(n ** 0.5) + 1):
        if n % i == 0 or n % (i + 2) == 0: return False
    return True

def ThreeDivisors(n):
    root = int(n ** 0.5)
    return root * root == n and isPrime(root)

def main():
    n = int(input())
    cnt = 0
    for i in range(2, int(n ** 0.5) + 1):
        if ThreeDivisors(i * i):
            cnt += 1
    print(cnt)

if __name__:
    main()